server/user/user_store.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package user
+
+import (
+	"github.com/mattermost/mattermost-plugin-msoffice/server/kvstore"
+)
+
+const UserKeyPrefix = "user_"
+
+type Store interface {
+	LoadRemoteUser(mattermostUserId string, ref interface{}) error
+	LoadMattermostUserId(remoteUserId string) (string, error)
+	Store(mattermostUserId string, user *User) error
+	Delete(mattermostUserId string) error
+}
+
+type store struct {
+	kv kvstore.KVStore
+}
+
+func NewStore(s kvstore.KVStore) Store {
+	return &store{
+		kv: kvstore.NewHashedKeyStore(s, UserKeyPrefix),
+	}
+}
+
+func (s *store) LoadRemoteUser(mattermostUserId string, ref interface{}) error {
+	return kvstore.LoadJSON(s.kv, mattermostUserId, ref)
+}
+
+func (s *store) LoadMattermostUserId(remoteUserId string) (string, error) {
+	data, err := s.kv.Load(remoteUserId)
```

Comment:
Do we want to prefix the `remoteUserId` with something, for the keys in the kv store? Along with every other user-specific key

Reply:
It is, see https://github.com/mattermost/mattermost-plugin-msoffice/pull/1/files#diff-4cd1f48867ecea93ce7781326fe70ca5R25